tools/convert_usernames_to_ids.py

```diff
@@ -0,0 +1,153 @@
+import os
+import json
+from collections import defaultdict
+import shutil
+
+
+def get_user_id_and_nickname(username_folder_path):
+
+    username = os.path.basename(username_folder_path)
+    json_file = os.path.join(username_folder_path, f"{username}.json")
+
+    if not os.path.exists(json_file):
+        return None, None
+
+    with open(json_file, 'r') as f:
+        user_file = json.load(f)
+        for story in user_file:
+            if 'user' in story and 'pk' in story['user'] and 'username' in story['user']:
+                return [str(story['user']['pk']), story['user']['username']]
+        raise ValueError('No ID/Nickname Found')
+
+
+def create_id_folder_mapping(base_folder):
+    id_to_path_mapping = defaultdict(list)
+    for username_folder in os.listdir(base_folder):
+        username_folder_path = os.path.join(base_folder, username_folder)
+        user_id, _ = get_user_id_and_nickname(username_folder_path)
+        if user_id:
+            id_to_path_mapping[user_id].append(username_folder_path)
+    return id_to_path_mapping
+
+
+def move_all_subfolders(source, destination):
+    for file in os.listdir(source):
```

Comment:
nit: I think here it's better to user "folder" instead "file", because you're skipping files and operating with folders